### PR TITLE
Revert signature change to IotNetworkInterface_t::create, doc updates

### PR DESCRIPTION
**Pull request by gkwicker**
PR #603 changed the signature of `IotNetworkInterface_t::create()`.  After review we've decided to revert this and keep it as it was to reduce the chance of incorrect configuration when multiple network interfaces exist on a system.  This change also addresses some documentation issues that were raised for PR #603 after it had been merged.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: libraries/standard/mqtt/include/types/iot_mqtt_types.h [hunk 1]
```diff
@@ -998,7 +998,7 @@ typedef struct IotMqttNetworkInfo
              * interface when creating a new network connection. It is only valid when
              * #IotMqttNetworkInfo_t::createNetworkConnection is `true`.
              */
-            IotNetworkServerInfo_t * pNetworkServerInfo;
```
Comment (abhidixi11) [inline libraries/standard/mqtt/include/types/iot_mqtt_types.h]: Line 948-957 are not needed and should be removed.

File: libraries/standard/mqtt/include/iot_mqtt.h [hunk 2]
```diff
@@ -807,8 +807,7 @@ const char * IotMqtt_OperationType( IotMqttOperationType_t operation );
  * @param[in] pTopicFilter The topic filter to check.
  * @param[in] topicFilterLength Length of `pTopicFilter`.
  * @param[out] pCurrentSubscription If a subscription is found, its details are
- * copied here. This output parameter is only valid if this function returns `true`.
- * Pass `NULL` to ignore.
+ * copied here. This output parameter is only valid if this function returns `true` (NULL to disable)
```
Comment (gordonwang0) [inline libraries/standard/mqtt/include/iot_mqtt.h]: Nit: period at end of comment, use code syntax for NULL: (`NULL` to disable).

File: libraries/standard/mqtt/include/types/iot_mqtt_types.h (hunk 1 above)
Comment (gkwicker) [reply libraries/standard/mqtt/include/types/iot_mqtt_types.h]: Good catch, removed.

File: libraries/standard/mqtt/include/iot_mqtt.h (hunk 2 above)
Comment (gkwicker) [reply libraries/standard/mqtt/include/iot_mqtt.h]: Added period here and changed to `NULL` throughout.

**Pull request merged**